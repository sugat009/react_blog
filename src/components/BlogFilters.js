import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTitleFilter, sortByAlphabet, sortByDate, setStartDate, setEndDate } from '../actions/filters';

export class BlogFilters extends React.Component {
    state = {
        calenderFocused:  null
    };
    onSortChange = (e) => {
        if (e.target.value === 'date') {
            this.props.sortByDate();
        } else if (e.target.value === 'alphabetically') {
            this.props.sortByAlphabet();
        }
    };
    onTextChange = (e) => {
        this.props.setTitleFilter(e.target.value);
    };
    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };
    onFocusChange = (calenderFocused) => {
        this.setState(() => ({ calenderFocused }));
    };
    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input 
                            className="text-input"
                            type="text"
                            placeholder="search blogs"
                            value={this.props.filters.title}
                            onChange={this.onTextChange}
                        />
                    </div>
                    <div className="input-group__item">
                        <select 
                            className="select"
                            value={this.props.filters.sortBy}
                            onChange={this.onSortChange}
                        >
                            <option value="alphabetically">Alphabetically</option>
                            <option value="date">Date</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker 
                            startDate = {this.props.filters.startDate}
                            endDate = {this.props.filters.endDate}
                            onDatesChange = {this.onDatesChange}
                            focusedInput = {this.state.calenderFocused}
                            onFocusChange = {this.onFocusChange}
                            numberOfMonths = {1}
                            isOutsideRange = {() => false}
                            showClearDates = {true}
                        />
                    </div>
                </div>
            </div>
        )
    };
};


const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
    setTitleFilter: (title) => dispatch(setTitleFilter(title)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAlphabet: () => dispatch(sortByAlphabet()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(BlogFilters);