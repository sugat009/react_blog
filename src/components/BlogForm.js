import React from 'react';
import moment from 'moment';

export default class BlogForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            title: props.blog ? props.blog.title : '',
            content: props.blog ? props.blog.content : '',
            createdAt: props.blog ? moment(props.blog.createdAt) : moment(0),
            errors: ''
        };
    };
    contentChange = (e) => {
        const content = e.target.value;
        this.setState({
            content
        });
    };
    onTitleChange = (e) => {
        const title = e.target.value;
        this.setState({
            title
        });
    };
    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.title || !this.state.content) {
            this.setState({
                errors: "Both title and content are required!"
            });
        } else {
            this.setState(() => ({ errors: ''}));
            if (moment(this.state.createdAt).isSame(moment(0))) {
                this.props.onSubmit({
                    title: this.state.title,
                    content: this.state.content,
                    createdAt: moment().valueOf()
                });
            } else {
                this.props.onSubmit({
                    title: this.state.title,
                    content: this.state.content
                });
            }
        }
    };
    render() {
        return (
            <form onSubmit={this.onSubmit} className="form">
                {this.state.errors && <p className="form__error">{this.state.errors}</p> }
                <input 
                    className="text-input"
                    type="text"
                    placeholder="Blog Title"
                    autoFocus
                    value={this.state.title}
                    onChange={this.onTitleChange}
                />
                <textarea 
                    className="text-area"
                    placeholder="Blog Content"
                    value={this.state.content}
                    onChange={this.contentChange}
                />
                <div>
                    <button className="button">Save Blog</button>
                </div>
            </form>
        )
    };
};