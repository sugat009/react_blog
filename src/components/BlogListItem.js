import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const BlogListItem = ({ id, title, content, createdAt, user }) => (
    <Link to={`/edit/${id}`} className="list-item">
        <div>
            <h3 className="list-item__title">{title}</h3>
            <span className="list-item__subtitle">{moment(createdAt).format("MMMM Do, YYYY")}</span>
        </div>
        <h3 className="list-item__data">{user.displayName}</h3>
    </Link>
);

export default BlogListItem;