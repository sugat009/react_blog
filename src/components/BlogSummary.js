import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectBlogs from '../selectors/blogs';
import selectBlogsTotal from '../selectors/blogs-total';

export const BlogSummary = ({ blogCount, blogsTotal }) => {
    const blogWord = blogCount === 1 ? 'Blog' : 'Blogs';
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{blogCount}</span> {blogWord} of <span>{blogsTotal}</span>.</h1>
                <div className="page-header__actions">
                    <Link to="/create" className="button">Add Blog</Link>
                    <Link to="/read" className="button">View Blogs</Link>
                </div>
            </div>
        </div>
    )
};

const mapStateToProps = (state) => {
    const visibleBlogs = selectBlogs(state.blogs, state.filters);
    return {
        blogCount: visibleBlogs.length,
        blogsTotal: selectBlogsTotal(visibleBlogs)
    }
};

export default connect(mapStateToProps)(BlogSummary);