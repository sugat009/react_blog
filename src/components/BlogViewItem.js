import React from 'react';
import moment from 'moment';

const BlogViewItem = () => (
    <div>
        <div>
            <h3 className="list-item__title">{title}</h3>
            <span className="list-item__subtitle">{moment(createdAt).format("MMMM Do, YYYY")}</span>
        </div>
        <h3 className="list-item__data">{user.displayName}</h3>
    </div>
);

export default BlogViewItem;
