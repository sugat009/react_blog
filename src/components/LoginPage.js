import React from 'react';
import { connect } from 'react-redux';
import { startLoginWithGoogle, startLoginWithFacebook } from '../actions/auth';

export class LoginPage extends React.Component {
    render() {
        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <h1 className="box-layout__title">React Blog</h1>
                    <p>Just React!!!</p>
                    <button className="button" onClick={this.props.startLoginWithGoogle}>Login With Google</button>
                    <button className="button button--fb" onClick={this.props.startLoginWithFacebook}>Login With Facebook</button>
                </div>
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch) => ({
    startLoginWithGoogle: () => dispatch(startLoginWithGoogle()),
    startLoginWithFacebook: () => dispatch(startLoginWithFacebook())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);