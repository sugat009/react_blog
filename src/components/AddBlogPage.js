import React from 'react';
import { connect } from 'react-redux';
import BlogForm from './BlogForm';
import { startAddBlog } from '../actions/blogs';


export class AddBlogPage extends React.Component {
    onSubmit = (blog) => {
        this.props.startAddBlog(blog);
        this.props.history.push("/");
    };
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add Blog</h1>
                    </div>
                </div>
                <div className="content-container">
                    <BlogForm onSubmit={this.onSubmit} />
                </div>
            </div>
        )
    };
};

const mapDispatchToProps = (dispatch) => ({
    startAddBlog: (blog) => dispatch(startAddBlog(blog))
});

export default connect(undefined, mapDispatchToProps)(AddBlogPage); // no mapStateToProps so undefined is used