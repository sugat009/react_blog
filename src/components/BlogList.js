import React from 'react';
import { connect } from 'react-redux';
import BlogListItem from './BlogListItem';
import getVisibleExpenses from '../selectors/blogs';

export class BlogList extends React.Component {
    render() {
        var user = JSON.parse(localStorage.getItem('user'));
        return (
            <div className="content-container">
                <div className="list-header">
                    <div className="show-for-mobile">Blogs</div>
                    <div className="show-for-desktop">Blog</div>
                    <div className="show-for-desktop">Author</div>
                </div>
                <div className="list-body">
                    {
                        this.props.blogs.length === 0 ? (
                            <div className="list-item list-item--message">
                                <span>No Blogs Yet</span>
                            </div>
                        ) : (
                        this.props.blogs.map((blog) => {
                                return <BlogListItem key = {blog.id} {...blog} user={user} />
                        })
                        )
                    }
                </div>
            </div>
        );
    };
};

const mapStateToProps = (state) => ({
    blogs: getVisibleExpenses(state.blogs, state.filters)
});

export default connect(mapStateToProps)(BlogList);