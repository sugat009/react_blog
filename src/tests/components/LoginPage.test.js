import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';


test('should render out LoginPage correctly', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});


test('should call startLogin on button click', () => {
    const startLoginWithGoogle = jest.fn();
    const wrapper = shallow(<LoginPage startLoginWithGoogle={startLoginWithGoogle} />);
    wrapper.find('button').first().simulate('click');
    expect(startLoginWithGoogle).toHaveBeenCalled();
});