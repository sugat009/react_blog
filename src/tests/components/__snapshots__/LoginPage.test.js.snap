// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render out LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      React Blog
    </h1>
    <p>
      Just React!!!
    </p>
    <button
      className="button"
    >
      Login With Google
    </button>
    <button
      className="button button--fb"
    >
      Login With Facebook
    </button>
  </div>
</div>
`;
